### PR TITLE
add buttons for setting different API endpoints

### PR DESCRIPTION
**Pull request by XertroV**
image
image

### CODE REVIEW

File: src/Settings/AdvancedParameters.as
Comment (dwhitacre) [inline src/Settings/AdvancedParameters.as:29]: is this needed, use default should reset to this?

File: src/Settings/AdvancedParameters.as
Comment (dwhitacre) [inline src/Settings/AdvancedParameters.as:36]: this api has not gone public and doesnt really need a dedicated button atm

**Review by dwhitacre — CHANGES_REQUESTED** [review]

File: src/Settings/AdvancedParameters.as
Comment (XertroV) [reply src/Settings/AdvancedParameters.as]: Commented for the moment until it's ready (sorry, shouldn't have added in the first place but didn't want to make the PR just for my API)

File: src/Settings/AdvancedParameters.as
Comment (XertroV) [reply src/Settings/AdvancedParameters.as:29]: Reset to default does, but also resets other things.

Comment (XertroV) [reply src/Settings/AdvancedParameters.as:29]: I thought it'd be better to have a button that explicitly does just that, both for consistency (there will be other buttons that set the URL), and because it's more obvious from a UX point of view how to undo changing the URL without affecting other settings.

File: src/Settings/AdvancedParameters.as
Comment (GreepTheSheep) [inline src/Settings/AdvancedParameters.as:32]: I'll prefer wait until it's ready to start merging this PR :)

**Review by GreepTheSheep — CHANGES_REQUESTED** [review]
Can you make a change to the variable (see comment)

File: src/Settings/AdvancedParameters.as
Comment (XertroV) [reply src/Settings/AdvancedParameters.as:32]: Sounds good

**Comment by XertroV** [comment]
> Can you make a change to the variable (see comment)

I'm not sure which comment you're referring to, or which variable. The only comment I saw you make was about Dan's API and waiting to merge. If you were talking about that, it's not obvious what change you're after (just uncomment again?).

**Comment by GreepTheSheep** [comment]
yeah this comment, forgot to sent it x)

**Comment by XertroV** [comment]
> yeah this comment, forgot to sent it x)

I still don't know what change you want. can you please just tell me?

Also, mb remove this from the 2.3.5 milestone? There is an unreleased fix that people are waiting for before doing more RMT, and it doesn't make sense to hold up releasing a fix to a game breaking bug because of a button to set a text value for an alt API.

File: src/Settings/AdvancedParameters.as
Comment (dwhitacre) [reply src/Settings/AdvancedParameters.as:29]: i dont agree nor disagree on your assessment, but am fine with keeping it

**Comment by dwhitacre** [comment]
My api is technically up and running and ready to go, but I am not sure I love the idea of making it a button in the menu. Although, I was happy to spend the time to setup and host a separate api cache for myself, friends, and associates; I dont really have the time to maintain it for a larger user base. I am totally fine for people to stumble upon my api cache via informal means or sharing and to use it regularly, but making it an option in the UI makes it too easily available to become something that people just use as a replacement for using the tmx api directly; and then something Id feel a larger responsibility towards maintaining.

Ultimately, I'd rather people host their own copy of https://github.com/dwhitacre/mx-api-cache if they want to have their own personal cache of tmx random maps preloaded, rather than me try to host a single instance for all users. Furthermore, ive licensed it as MIT so if people wanted to implement the same code or ideas straight into their own services, id happily support and help as well.

**Pull request merged**
